Samples.Swift/Samples.Swift/ViewControllers/AddItemViewConroller/AddItemViewController.swift

```diff
@@ -0,0 +1,194 @@
+    button.layer.cornerRadius = 8
+    button.backgroundColor = .systemBlue
+    button.tintColor = .white
+    button.frame.size.height = 40
+    button.titleLabel?.font = .boldSystemFont(ofSize: 16)
+    button.setTitle("Add", for: .normal)
+    button.addTarget(self, action: #selector(addItemTapped), for: .touchUpInside)
+    return button
+  }()
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.gray.withAlphaComponent(0.5)
+    configure()
+  }
+
+  lazy var stackView: UIStackView = {
+    let stackView = UIStackView(
+      arrangedSubviews: [
+        createItemLabel,
+        nameField,
+        amountField,
+        quantityField,
+        taxAmountField,
+        addItemButton
+      ]
+    )
+    stackView.axis = .vertical
+    stackView.spacing = 10
+    stackView.distribution = .fill
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.layoutMargins = UIEdgeInsets(
+      top: 5.0,
+      left: 16.0,
+      bottom: 16.0,
+      right: 16.0
+    )
+    return stackView
+  }()
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+    self.nameField.text = ""
+    self.amountField.text = ""
+    self.quantityField.text = ""
+    self.taxAmountField.text = ""
+
+    createItemLabel.text = "Create Item"
+    addItemButton.setTitle("Add", for: .normal)
+  }
+
+  init(
+    item: PurchaseUnit.Item
+  ) {
+    super.init(nibName: nil, bundle: nil)
+    self.nameField.text = item.name
+    self.amountField.text = item.unitAmount.value
+    self.quantityField.text = item.quantity
+    self.taxAmountField.text = item.tax?.value
+
+    createItemLabel.text = "Edit Item"
+    addItemButton.setTitle("Save", for: .normal)
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  @objc
+  func addItemTapped() {
+    if nameField.text == "" || amountField.text == "" || quantityField.text == "" || taxAmountField.text == "" {
+      let ac = UIAlertController(title: "Cannot add item", message: "Please ensure all item fields are filled out", preferredStyle: .alert)
+      ac.addAction(UIAlertAction(title: "Ok", style: .default))
+      self.present(ac, animated: true)
+    }
+    else {
+      let item: PurchaseUnit.Item = PurchaseUnit.Item(
+        name: nameField.text ?? "",
+        unitAmount: UnitAmount(
+          currencyCode: .usd,
+          value: amountField.text ?? ""
+        ),
+        quantity: quantityField.text ?? "",
+        tax: PurchaseUnit.Tax(
+          currencyCode: .usd,
+          value: taxAmountField.text ?? ""
+        )
+      )
+      delegate?.didTapSaveWithItem(item: item, at: index ?? 0)
+      self.dismiss(animated: true)
+    }
+  }
+
+  override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+    super.touchesBegan(touches, with: event)
+
+    let touch = touches.first
+    guard let location = touch?.location(in: self.view) else { return }
```

Comment:
can collapse this into 
`guard let location = touches.first?.location(in: self.view) else { return }`